pkg/virt-api/webhooks/validating-webhook/admitters/vms-admitter_test.go

```diff
@@ -1258,33 +1260,75 @@ var _ = Describe("Validating VM Admitter", func() {
 
 	Context("Instancetype", func() {
 		var (
-			vm *v1.VirtualMachine
+			vm               *v1.VirtualMachine
+			testInstancetype *v1beta1.VirtualMachineInstancetype
+			testPreference   *v1beta1.VirtualMachinePreference
 		)
 
 		BeforeEach(func() {
-			vmi := api.NewMinimalVMI("testvmi")
-			vm = &v1.VirtualMachine{
-				Spec: v1.VirtualMachineSpec{
-					Instancetype: &v1.InstancetypeMatcher{
-						Name: "test",
-						Kind: instancetypeapi.SingularResourceName,
+			vm = libvmi.NewVirtualMachine(
+				libvmi.New(
+					libvmi.WithNamespace("test"),
+				),
+				libvmi.WithInstancetype("test"),
+				libvmi.WithPreference("test"),
+			)
+
+			testInstancetypeClients := fakeclientset.NewSimpleClientset().InstancetypeV1beta1()
+
+			testInstancetypeClient := testInstancetypeClients.VirtualMachineInstancetypes(vm.Namespace)
+			virtClient.EXPECT().VirtualMachineInstancetype(gomock.Any()).Return(testInstancetypeClient).AnyTimes()
+
+			//TODO(lyarwood): Add WithNamespace to libinstancetype.builder and use here
+			testInstancetype = &instancetypev1beta1.VirtualMachineInstancetype{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       instancetypeapi.SingularResourceName,
+					APIVersion: instancetypev1beta1.SchemeGroupVersion.String(),
+				},
```

Comment:
Will it fail when you leave out `TypeMeta`?

Reply:
Yup sorry that's populated for us with the fakeclients.